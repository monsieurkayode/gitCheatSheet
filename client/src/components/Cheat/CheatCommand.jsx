import React from 'react';
import PropTypes from 'prop-types';

const CheatCommand = ({ command }) => {
  const [keyword, ...body] = command.split(' ');

  return (
    <pre className="command">
      <code>&#36;&nbsp;
        <span className="git">{keyword}</span>&nbsp;{body.join(' ')}
      </code>
    </pre>
  );
};

CheatCommand.propTypes = {
  command: PropTypes.string.isRequired
};

export default CheatCommand;
