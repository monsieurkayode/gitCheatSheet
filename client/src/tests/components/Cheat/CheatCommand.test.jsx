import React from 'react';
import { shallow } from 'enzyme';
import CheatCommand from '../../../components/Cheat/CheatCommand';

const setup = () => {
  const props = {
    command: 'git merge',
    onCopy: jest.fn(),
    id: '1',
  };

  const wrapper = shallow(<CheatCommand {...props} />);
  return { wrapper, props };
};

describe('CheatCard Component', () => {
  const { wrapper, props } = setup();

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onCopy', () => {
    wrapper.find('.command').simulate('click');
    expect(props.onCopy).toHaveBeenCalled();
  });
});
