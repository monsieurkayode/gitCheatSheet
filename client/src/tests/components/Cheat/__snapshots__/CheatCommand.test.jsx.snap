// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheatCard Component should render correctly 1`] = `
<pre
  className="command"
  onClick={[Function]}
>
  <CopyToClipboard
    text="git merge"
  >
    <code>
      $ 
      <span
        className="git"
      >
        git
      </span>
       
      merge
    </code>
  </CopyToClipboard>
</pre>
`;
